etsin_finder/frontend/__tests__/combined/qvain/qvain.fileForm.test.jsx

```diff
@@ -0,0 +1,297 @@
+    const selectFileType = async label => {
+      return selectOption('file-form-file-type', label)
+    }
+
+    it('should allow selecting use category', async () => {
+      await render()
+      await selectUseCategory('Documentation')
+      await expectSave({
+        title: 'testfile',
+        description: 'File',
+        useCategory: 'http://uri.suomi.fi/codelist/fairdata/use_category/code/documentation',
+        fileType: undefined,
+      })
+    })
+
+    it('should fail to save without use category', async () => {
+      await render()
+      await expectError('qvain.validationMessages.files.file.useCategory.required')
+    })
+
+    it('should allow selecting file type', async () => {
+      await render()
+      await selectUseCategory('Documentation')
+      await selectFileType('Audiovisual')
+      await expectSave({
+        title: 'testfile',
+        description: 'File',
+        useCategory: 'http://uri.suomi.fi/codelist/fairdata/use_category/code/documentation',
+        fileType: 'http://uri.suomi.fi/codelist/fairdata/file_type/code/audiovisual',
+      })
+    })
+
+    it('should allow changing title', async () => {
+      await render()
+      await selectUseCategory('Documentation')
+      wrapper.find('input#file-form-title').simulate('change', { target: { value: 'New Title' } })
+      await expectSave({
+        title: 'New Title',
+        description: 'File',
+        useCategory: 'http://uri.suomi.fi/codelist/fairdata/use_category/code/documentation',
+        fileType: undefined,
+      })
+    })
+
+    it('should fail to save without title', async () => {
+      await render()
+      await selectUseCategory('Documentation')
+      wrapper.find('input#file-form-title').simulate('change', { target: { value: '' } })
+      await expectError('qvain.validationMessages.files.file.title.required')
+    })
+
+    it('should allow changing description', async () => {
+      await render()
+      await selectUseCategory('Documentation')
+      wrapper
+        .find('textarea#file-form-description')
+        .simulate('change', { target: { value: 'This is a great description' } })
+      await expectSave({
+        title: 'testfile',
+        description: 'This is a great description',
+        useCategory: 'http://uri.suomi.fi/codelist/fairdata/use_category/code/documentation',
+        fileType: undefined,
+      })
+    })
+  })
+
+  describe('DirectoryForm', () => {
+    const selectUseCategory = label => {
+      return selectOption('directory-form-use-category', label)
+    }
+
+    it('should allow selecting use category', async () => {
+      await render(true)
+      await selectUseCategory('Documentation')
+      await expectSave({
+        title: 'testdir',
+        useCategory: 'http://uri.suomi.fi/codelist/fairdata/use_category/code/documentation',
+      })
+    })
+
+    it('should fail to save when use category is not set', async () => {
+      await render(true)
+      // await expectSave(undefined)
```

Comment:
clear commented code. Also three flushPromise calls in a row would deserve an explanation/comment

Reply:
those are from debugging, forgot to remove them (that problem was fixed by adding a wrapper.update to expectSave)